cli/ts/commands/signup.ts

```diff
@@ -57,19 +70,24 @@ export const signup = async (
 
     logYellow(quiet, info(`Transaction hash: ${tx.hash}`));
 
-    if (receipt.status !== 1) logError("The transaction failed");
+    if (receipt?.status !== 1) {
+      logError("The transaction failed");
+    }
+
     const iface = maciContract.interface;
 
     // get state index from the event
-    if (receipt && receipt.logs) {
-      stateIndex = iface.parseLog(receipt.logs[0]).args[0];
+    if (receipt?.logs) {
+      const [log] = receipt.logs;
+      const { args } = iface.parseLog(log as unknown as { topics: string[]; data: string }) || { args: [] };
+      [stateIndex] = args;
       logGreen(quiet, success(`State index: ${stateIndex.toString()}`));
     } else {
       logError("Unable to retrieve the transaction receipt");
     }
-  } catch (error: any) {
-    logError(error.message);
+  } catch (error) {
+    logError((error as Error).message);
   }
 
-  return stateIndex.toString();
+  return stateIndex ? stateIndex.toString() : "";
```

Comment:
Just curious why we add `.toString()` here, since I think the `stateIndex` itself is a `string` already?

Reply:
I didn't change any logic here. This PR is related to lint fixes.